code/modules/shuttle/mini_dropship.dm

```diff
@@ -10,33 +10,184 @@
+
+///Change the icons of the space turf in the transit
+/obj/machinery/computer/camera_advanced/shuttle_docker/minidropship/proc/set_transit_turf(in_atmos = FALSE)
+	var/turf_type = in_atmos ? /turf/open/space/transit/atmos : /turf/open/space/transit
+	var/former_type = in_atmos ? /turf/open/space/transit : /turf/open/space/transit/atmos
+	for(var/turf/T AS in shuttle_port.assigned_transit.reserved_area.reserved_turfs)
+		if(istype(T, former_type))
+			T.ChangeTurf(turf_type, turf_type)
+
+///The action of taking off and sending the shuttle to the atmosphere
+/obj/machinery/computer/camera_advanced/shuttle_docker/minidropship/proc/take_off()
+	shuttle_port = SSshuttle.getShuttle(shuttleId)
+	/*if(!(shuttle_port.shuttle_flags & GAMEMODE_IMMUNE) && world.time < SSticker.round_start_time + SSticker.mode.deploy_time_lock)
+		to_chat(ui_user, "<span class='warning'>The engines are still refueling.</span>")
+		return*/
+	shuttle_port.shuttle = src
+	if(fly_state == SHUTTLE_ON_GROUND)
+		next_fly_state = SHUTTLE_IN_ATMOSPHERE
+		shuttle_port.callTime = SHUTTLE_TAKEOFF_GROUND_CALLTIME
+	else
+		to_transit = TRUE
+		shuttle_port.callTime = SHUTTLE_TAKEOFF_SHIP_CALLTIME
+		next_fly_state = SHUTTLE_IN_SPACE
+		destination_fly_state = SHUTTLE_IN_ATMOSPHERE
+	SSshuttle.moveShuttleToTransit(shuttleId, TRUE)
+
+///The action of sending the shuttle back to its shuttle port on main ship
+/obj/machinery/computer/camera_advanced/shuttle_docker/minidropship/proc/return_to_ship()
+	shuttle_port = SSshuttle.getShuttle(shuttleId)
+	shuttle_port.shuttle = src
+	to_transit = TRUE
+	next_fly_state = SHUTTLE_IN_SPACE
+	destination_fly_state = SHUTTLE_ON_SHIP
+	if(!origin_port_id)
+		return
+	open_prompt = FALSE
+	remove_eye_control(ui_user)
+	SSshuttle.moveShuttle(shuttleId, origin_port_id, TRUE)
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/minidropship/ui_state(mob/user)
+	return GLOB.dropship_state
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/minidropship/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+
+	if(!ui)
+		ui_user = user
+		ui = new(user, src, "Minidropship", name)
+		ui.open()
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/minidropship/ui_data(mob/user)
+	. = list()
+	.["fuel_left"] = fuel_left
+	.["fuel_max"] = fuel_max
+	.["fly_state"] = fly_state
+	.["take_off_locked"] = !(fly_state == SHUTTLE_ON_GROUND || fly_state == SHUTTLE_ON_SHIP)
+	.["return_to_ship_locked"] = (fly_state != SHUTTLE_IN_ATMOSPHERE)
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/minidropship/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	switch(action)
+		if("take_off")
+			take_off()
+		if("return_to_ship")
+			return_to_ship()
+
+/datum/action/innate/shuttledocker_land
+	name = "Land"
+	action_icon = 'icons/mecha/actions_mecha.dmi'
+	action_icon_state = "land"
+
+/datum/action/innate/shuttledocker_land/Activate()
+	if(QDELETED(target) || !isliving(target))
```

Comment:
Why the qdeleted check

Reply:
it was there in the action i took in example (the one to place the designated landing spot)

Reply:
it seems redundant? or does it just not call parent for those checks?